eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -276,8 +277,12 @@ public function deleteVersion($contentId, $versionNo)
     /**
      * {@inheritdoc}
      */
-    public function listVersions($contentId, $status = null, $limit = -1)
+    public function listVersions($contentId, $status = null, $limit = self::DEFAULT_LIST_VERSIONS_LIMIT)
     {
+        if ($limit !== self::DEFAULT_LIST_VERSIONS_LIMIT) {
+            return $this->persistenceHandler->contentHandler()->listVersions($contentId, $status, $limit);
+        }
```

Comment:
Question is what value Admin UI uses for limit. If not always default, then we are not caching what we need to cache ;)

Reply:
I think it's correct, AdminUI uses default value: https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/lib/UI/Dataset/VersionsDataset.php#L47